### PR TITLE
KVC-LB131.LV_R_GLU

### PR DESCRIPTION
**Pull request by kvcalvin**
Opening a pull request for LB131.LV_R_GLU code.

### CODE REVIEW

File: R/constants.R [hunk 1]
```diff
@@ -23,6 +23,7 @@ FUTURE_YEARS <- seq( 2015, 2100, 5 )
 # aglu constants
 AGLU_HISTORICAL_YEARS <- 1971:2010
 FAO_HISTORICAL_YEARS <- 1961:2011
+X_GTAP_HISTORICAL_YEAR <- "X2000"
```
Comment (bpbond) [inline R/constants.R]: Change this to just `GTAP_HISTORICAL_YEAR <- 2000` and...

File: R/zchunk_LB131.LV_R_GLU.R [hunk 2]
```diff
@@ -29,42 +30,37 @@ module_aglu_LB131.LV_R_GLU_DISABLED <- function(command, ...) {
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     L100.GTAP_LV_milUSD <- get_data(all_data, "L100.GTAP_LV_milUSD")
     L122.LC_bm2_R_HarvCropLand_Yh_GLU <- get_data(all_data, "temp-data-inject/L122.LC_bm2_R_HarvCropLand_Yh_GLU")
```
Comment (bpbond) [inline R/zchunk_LB131.LV_R_GLU.R]: ...and if data coming in here with "X" years, strip them off here up top. When we remove temp data injection, you'll just have to remove a single line.

File: R/zchunk_LB131.LV_R_GLU.R [hunk 3]
```diff
@@ -29,42 +30,37 @@ module_aglu_LB131.LV_R_GLU_DISABLED <- function(command, ...) {
...
+      mutate(value = value * CONV_2001_1975_USD) %>%                                                    # Convert to 1975$
+      group_by(GCAM_region_ID, GLU) %>%                                                                 # Group by GCAM_region_ID and GLU
+      summarize(value = sum(value)) %>%                                                                 # Aggregate value to GCAM region and GLU
+      rename(LV_milUSD75 = value) ->                                                                    # Rename column to what is used in old data system
+      LV_R_GLU
+    
+    # Prepare land area for use in compuating land value
+    L122.LC_bm2_R_HarvCropLand_Yh_GLU %>%
+      gather(year, value, -GCAM_region_ID, -GLU, -Land_Type) %>%                                      # Convert to long format
+      filter(year == X_GTAP_HISTORICAL_YEAR) %>%                                                      # Only use cropland area from the GTAP historical year
```
Comment (bpbond) [inline R/zchunk_LB131.LV_R_GLU.R]: GTAP_HISTORICAL_YEAR

File: R/zchunk_LB131.LV_R_GLU.R (hunk 2 above)
Comment (bpbond) [inline R/zchunk_LB131.LV_R_GLU.R]: I.e., change line 32 to:

```R
get_data(all_data, "temp-data-inject/L122.LC_bm2_R_HarvCropLand_Yh_GLU") %>%
      # Two temporary lines for data injection
      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
      mutate(year = as.integer(substr(year, 2, 5))) ->
      L122.LC_bm2_R_HarvCropLand_Yh_GLU
```

File: R/zchunk_LB131.LV_R_GLU.R (hunk 3 above)
Comment (bpbond) [inline R/zchunk_LB131.LV_R_GLU.R]: And remove line 45

**Comment by bpbond** [comment]
@kvcalvin I made those changes so that @swaldhoff can review and we can get this merged.

**Comment by kvcalvin** [comment]
Thanks!

**Pull request merged**